spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -277,13 +292,15 @@ private void prepareEvaluationContext() {
 			}
 		}
 		else {
-			AnnotatedMethodFilter filter = new AnnotatedMethodFilter(this.annotationType, this.methodName,
-					this.requiresReply);
-			Assert.state(canReturnExpectedType(filter, targetType, context.getTypeConverter()),
+			AnnotatedMethodFilter annotatedMethodFilter = new AnnotatedMethodFilter(this.annotationType,
+					this.methodName, this.requiresReply);
+			Assert.state(canReturnExpectedType(annotatedMethodFilter, targetType, context.getTypeConverter()),
 					"Cannot convert to expected type (" + this.expectedType + ") from " + this.method);
-			context.registerMethodFilter(targetType, filter);
+			context.registerMethodFilter(targetType, annotatedMethodFilter);
+			declaredMethodResolver.registerMethodFilter(targetType, annotatedMethodFilter);
 		}
 		context.setVariable("target", this.targetObject);
+		context.setMethodResolvers(Collections.singletonList(declaredMethodResolver));
```

Comment:
It's extremely low risk (the FB currently doesn't support customizing the method resolver) but let's say (for some reason) the user customizes the factory bean to create a context with some custom method resolver.

For this use case, we will override that customization - it's ok of course, because we need this level of control - but we should at least log that we have done so.